syntax = "proto3";
package protos.connectors.assets;

import "google/protobuf/wrappers.proto";
import "protos/base.proto";
import "protos/connectors/connector.proto";

message PostgresDatabaseAssetModel {
  google.protobuf.StringValue database = 1;
}

message PostgresDatabaseAssetOptions {
  repeated string databases = 1;
}

message PostgresAssetModel {
  google.protobuf.UInt64Value id = 1;
  Source connector_type = 2;
  ConnectorMetadataModelType type = 3;
  sfixed64 last_updated = 4;
  oneof asset {
    PostgresDatabaseAssetModel postgres_database = 5;
  }
}

message PostgresAssets {
  repeated PostgresAssetModel assets = 1;
}
